src/odemis/driver/tmcm.py

```diff
@@ -1278,6 +1305,23 @@ def _requestRefSwitch(self, axis):
             return
 
         with self._refswitch_lock:
+            # Set _leds_on attribute before closing shutters to make sure they are not
+            # opened again in a concurrent thread
+            self._leds_on = True  # do this before closing shutters
+            # Close shutters
+            shutter_move = {}
+            for channel, val in self._led_prot_do.items():
+                # TODO: is this safe (are we sure .position is up to date)?
```

Comment:
I think in theory, there could be simultaneously a call to reference an axis, and a call to move a DO axis. I think that could lead to either the DO axis not being restored to the expected position at the end of referencing, or (more dangerous), the DO axis being moved while ref leds are on.
There are two (relatively obvious) ways to handle it:
 * take the refswitch_lock when starting the move of a DO axis (and checking leds_on)
 * add extra dependencies on the reference moves, so that if the move involves an axis with a refswitch, it also depends (as in "the dependencies of _executor.submitf()") on the _led_prot_do axes.